### PR TITLE
Pass num_items_in_batch directly to loss computation

### LINKED ISSUE #36744: num_items_in_batch unexpected in vision encoder decoder
### System Info

Image

### Who can help?

_No response_

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

do forward on vision encoder decoder like donut.

### Expected behavior

model get trained

Image

### PR DESCRIPTION
**Pull request by eljandoubi**
# What does this PR do?

Fixes #36744

Models:

- vision models: @amyeroberts, @qubvel

### CODE REVIEW

File: src/transformers/models/vision_encoder_decoder/modeling_vision_encoder_decoder.py [hunk 1]
```diff
@@ -638,8 +641,12 @@ def forward(
...
         if labels is not None:
             logits = decoder_outputs.logits if return_dict else decoder_outputs[0]
-            loss_fct = CrossEntropyLoss()
-            loss = loss_fct(logits.reshape(-1, self.decoder.config.vocab_size), labels.reshape(-1))
+
+            loss = fixed_cross_entropy(
+                logits.reshape(-1, self.decoder.config.vocab_size),
+                labels.reshape(-1),
+                num_items_in_batch=num_items_in_batch,
+            )
```
Comment (qubvel) [inline src/transformers/models/vision_encoder_decoder/modeling_vision_encoder_decoder.py:649]: We can use instead, see llama for example. All reshape/view ops will happen under the hood
```python
loss = self.loss_function(logits=logits, labels=labels, vocab_size=self.config.vocab_size, num_items_in_batch=num_items_in_batch)
```

**Review by qubvel — COMMENTED** [review]
Thank you @eljandoubi for opening a PR and fixing the issue! Just a small nit

File: src/transformers/models/vision_encoder_decoder/modeling_vision_encoder_decoder.py (hunk 1 above)
Comment (eljandoubi) [reply src/transformers/models/vision_encoder_decoder/modeling_vision_encoder_decoder.py]: @qubvel  like this ?

Comment (qubvel) [reply src/transformers/models/vision_encoder_decoder/modeling_vision_encoder_decoder.py]: A bit different, there is no need to make view while passing to the function, see my example above

Comment (eljandoubi) [reply src/transformers/models/vision_encoder_decoder/modeling_vision_encoder_decoder.py]: @qubvel  I see

Comment (eljandoubi) [reply src/transformers/models/vision_encoder_decoder/modeling_vision_encoder_decoder.py]: @qubvel  I've updated my code, what do you think about it now?

**Review by ArthurZucker — APPROVED** [review]

**Comment by qubvel** [comment]
Thanks @eljandoubi
